### PR TITLE
[CPU] Added dynamism support for StridedSlice

### PR DESCRIPTION
**Pull request by a-sidorova**
### Details:
 - *Added dynamism support for StridedSlice*
 - *Added tests for dynamism*

### Tickets:
 - *70326*

### CODE REVIEW

File: inference-engine/src/mkldnn_plugin/nodes/mkldnn_strided_slice_node.h [hunk 1]
```diff
@@ -24,58 +25,73 @@ class MKLDNNStridedSliceNode : public MKLDNNNode {
...
+            InferenceEngine::SizeVector dstIndices;
+
+            size_t nThreads = 0lu;
+            size_t nDimsForWork = 0lu;
+            size_t workAmount = 0lu;
+            size_t lastDstDim = 0lu;
+            size_t srcShift = 0lu;
+
+            bool isOptimized = false;
+        } params;
```
Comment (maxnick) [inline inference-engine/src/mkldnn_plugin/nodes/mkldnn_strided_slice_node.h:84]: As far as I understand only part of this params data is used in the exec call. Do we really need to store all these data, especially attrs inside the executor objects?

Comment (a-sidorova) [reply inference-engine/src/mkldnn_plugin/nodes/mkldnn_strided_slice_node.h:84]: No, we don't. I left only necessary attributes for execution and created structure with useful parameters (strides, dims, etc) to comfortable work inside executor.
Good catch! Thank you!

File: inference-engine/src/mkldnn_plugin/nodes/mkldnn_strided_slice_node.h [hunk 2]
```diff
@@ -24,58 +25,73 @@ class MKLDNNStridedSliceNode : public MKLDNNNode {
...
+        InferenceEngine::SizeVector strideDims;
+
         bool equalDims = false;
-        bool parametersAreConstant = true;
-    } params;
+        size_t dataSize = 1lu;
+    } attrs;
+
+    struct StridedSliceExecutor {
+        StridedSliceExecutor(const StridedSliceAttributes& attrs, const InferenceEngine::SizeVector& srcDims, const InferenceEngine::SizeVector& dstDims);
```
[the remark this answers is not in the document]
Comment (a-sidorova) [reply inference-engine/src/mkldnn_plugin/nodes/mkldnn_strided_slice_node.h:57]: Done, thanks

File: inference-engine/tests/functional/plugin/cpu/shared_tests_instances/single_layer_tests/strided_slice.cpp [hunk 3]
```diff
@@ -12,7 +12,7 @@ using namespace LayerTestsDefinitions;
 namespace {
 
 std::vector<StridedSliceSpecificParams> ss_only_test_cases = {
-        StridedSliceSpecificParams{ { 16 }, { 4 }, { 12 }, { 1 },
+        StridedSliceSpecificParams{ { 16 }, { 4 }, { 10 }, { 1 },
```
[the remark this answers is not in the document]
Comment (a-sidorova) [reply inference-engine/tests/functional/plugin/cpu/shared_tests_instances/single_layer_tests/strided_slice.cpp:15]: I thought I saw similar test when output dim is [8]. But now I don't see (may be it seemed to me during debug). So I returned

**Review by maxnick — APPROVED** [review]

**Review by dmitry-gorokhov — APPROVED** [review]

**Pull request merged**